mesonbuild/coredata.py

```diff
@@ -629,7 +634,7 @@ def set_option(self, key: OptionKey, value) -> None:
                 value = self.sanitize_dir_option_value(prefix, key, value)
 
         try:
-            self.options[key].set_value(value)
+            self.options[key].set_value(value, True)
```

Comment:
AFAIK `set_option()` is called for `project(..., default_options: ...)` too, and in that case you still want to use auto_feature.